### DISCUSSION TITLE
如何爬详情页呢

### DISCUSSION THREAD

**Discussion by cxjhk** [coder-hxl/x-crawl#127] [General]
先爬列表数据，然后获取到每个item，点击item进去获取详情信息后，在点击下一个item

这个如何实现呢

**Comment by coder-hxl** [comment]
通过 crawlPage API 可实现
- https://coder-hxl.github.io/x-crawl/cn/guide/crawl-page
- https://coder-hxl.github.io/x-crawl/cn/api/crawl-page

**Comment by cxjhk** [comment]
循环多个crawlPage爬虫，爬完以后如何关闭browser呢

**Reply by coder-hxl** [reply]
这里 https://coder-hxl.github.io/x-crawl/cn/guide/crawl-page#crawl-the-page

**Comment by cxjhk** [comment]
爬分页数据，循环很多次，每次循环里都是crawlApp.crawlPage去爬页面，这样可行么，爬完了，如何关闭浏览器

**Comment by cxjhk** [comment]
还是说爬分页的话，每次都去createCrawl一个浏览器

**Comment by coder-hxl** [comment]
两种办法都可以

**Comment by coder-hxl** [comment]
用第一种比较好些

**Reply by cxjhk** [reply]
那第一种 完了以后 如何关闭browser呢

**Reply by coder-hxl** [reply]
第一种不用关闭 browser ，等全部爬完再关闭就行了。循环结束再关闭

**Reply by cxjhk** [reply]
哦哦，那我爬完了直接关闭了page

**Reply by coder-hxl** [reply]
也可以在 crawlPage 里面循环爬数据，就不用开那么多个页面了

**Reply by cxjhk** [reply]
在crawlPage里面重新newPage么

**Comment by cxjhk** [comment]
爬了item还要打开详情，详情爬完了业务才完

**Comment by coder-hxl** [comment]
不用，在一个页面操作，拿完一个页面再点击下一个分页，重复这个操作

**Reply by cxjhk** [reply]
爬vue的站，里面还有debug，等待元素出现的时候，一直获取不到元素，有解决办法吗

**Reply by coder-hxl** [reply]
可以去 https://pptr.dev/ 找一下
